redash/cli/organization.py

```diff
@@ -12,7 +13,7 @@ def set_google_apps_domains(domains):
     """
     Sets the allowable domains to the comma separated list DOMAINS.
     """
-    organization = models.Organization.query.first()
+    organization = models.db.session.scalar(select(models.Organization))
```

Comment:
session's scalar executes `Result.scalar` function, which returns first row
https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.scalar

Reply:
I checked the page. According to it, it looks like that it returns `the first column of the first row`...

> Fetch the first column of the first row, and close the result set.

What do you think about this?

Reply:
but `scalar()` returns an object of model type. we're getting an expected result

Reply:
So that's a different behavior than what's written in the documentation, right? Should Redash rely on it?

Reply:
the only exception is that scalar fails if amount of results is bigger than one

Reply:
the one it returns same as first, but ready object instead of dict

Reply:
> scalar returns same as first, but ready object instead of dict

Umm... it still shows that it returns `the first column of the first row` in the document.
However, you claims that `Result.scalar` returns the first `object`, right? So, is the API document of SQLAlchemy 2.0 not correct?

Reply:
scalar method returns different results depending on context. for session and connection it returns python object, for resultset it returns first value
```
method [sqlalchemy.engine.Connection.](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Connection)scalar(statement: [Executable](https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.Executable), parameters: _CoreSingleExecuteParams | None = None, *, execution_options: CoreExecuteOptionsParameter | None = None) → Any
Executes a SQL statement construct and returns a scalar object.
```

Reply:
I see. I checked this as well.
- https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.scalar

And `test_set_google_apps_domains` in test_cli.py passed. OK.